src/jobs/keys-update/keys-update.service.ts

```diff
@@ -90,119 +86,142 @@ export class KeysUpdateService {
 
-  /**
-   * Updates prometheus metrics
-   */
-  private setMetrics() {
-    // common metrics
-    if (this.lastTimestampSec) {
-      this.prometheusService.registryLastUpdate.set(this.lastTimestampSec);
-    }
-    if (this.lastBlockNumber) {
-      this.prometheusService.registryBlockNumber.set(this.lastBlockNumber);
+    // TODO: what will happen if module исчез из списка
+    // TODO: is it correct that i use here modules from blockchain instead of storage
+
+    if (!this.modulesWereNotDeleted(modules, storageModules)) {
+      const error = new Error('Modules list is wrong');
+      this.logger.error(error);
+      process.exit(1);
     }
 
-    // curated metrics
-    this.setCuratedMetrics();
-  }
+    await this.entityManager.transactional(
+      async () => {
+        // Update el meta in db
+        await this.elMetaStorage.update(currElMeta);
 
-  private setCuratedMetrics() {
-    if (this.curatedNonce) {
-      this.prometheusService.registryNonce.set({ srModuleId: 1 }, this.curatedNonce);
-    }
-    this.setCuratedOperatorsMetric();
+        for (const module of modules) {
+          const moduleInstance = this.stakingRouterService.getStakingRouterModuleImpl(module.type);
+
+          // At the moment lets think that for all modules it is possible to make decision base on nonce value
+          const currNonce = await moduleInstance.getCurrentNonce(module.stakingModuleAddress, currElMeta.hash);
+          const moduleInStorage = await this.srModulesStorage.findOneById(module.id);
 
-    this.logger.log('Curated Module metrics updated');
+          // now updating decision should be here moduleInstance.updateKeys
+          if (moduleInStorage && moduleInStorage.nonce === currNonce) {
+            // nothing changed, don't need to update
+            // TODO: add log
+            return;
+          }
+
+          await this.srModulesStorage.store(module, currNonce);
+          await moduleInstance.update(module.stakingModuleAddress, currElMeta.hash);
+        }
+      },
+      { isolationLevel: IsolationLevel.READ_COMMITTED },
+    );
+
+    return currElMeta;
   }
 
-  private setCuratedOperatorsMetric() {
-    this.prometheusService.registryNumberOfKeysBySRModuleAndOperator.reset();
-
-    this.curatedOperators.forEach((operator) => {
-      this.prometheusService.registryNumberOfKeysBySRModuleAndOperator.set(
-        {
-          operator: operator.index,
-          srModuleId: 1,
-          used: 'true',
-        },
-        operator.usedSigningKeys,
-      );
+  /**
+   * Update prometheus metrics of staking modules
+   */
+  public async updateMetrics() {
+    await this.entityManager.transactional(
+      async () => {
+        const stakingModules = await this.stakingRouterService.getStakingModules();
+        const elMeta = await this.stakingRouterService.getElBlockSnapshot();
+
+        if (!elMeta) {
+          return;
```

Comment:
I think we need to write some logs here.

Reply:
added in improvement PR